presto-main/src/main/java/com/facebook/presto/eventlistener/EventListenerManager.java

```diff
@@ -57,17 +71,38 @@ public void addEventListenerFactory(EventListenerFactory eventListenerFactory)
         }
     }
 
-    public void loadConfiguredEventListener()
-            throws Exception
+    public void loadConfiguredEventListeners()
     {
-        if (EVENT_LISTENER_CONFIGURATION.exists()) {
-            Map<String, String> properties = loadProperties(EVENT_LISTENER_CONFIGURATION);
-            checkArgument(
-                    !isNullOrEmpty(properties.get(EVENT_LISTENER_PROPERTY_NAME)),
-                    "Access control configuration %s does not contain %s",
-                    EVENT_LISTENER_CONFIGURATION.getAbsoluteFile(),
-                    EVENT_LISTENER_PROPERTY_NAME);
-            loadConfiguredEventListener(properties);
+        checkState(loading.compareAndSet(false, true), "Event listeners already loaded");
+        List<File> configFiles = this.configFiles;
+        if (configFiles.isEmpty()) {
+            if (!EVENT_LISTENER_CONFIGURATION.exists()) {
+                return;
+            }
+            configFiles = ImmutableList.of(EVENT_LISTENER_CONFIGURATION);
+        }
+        configFiles.forEach(this::createEventListener);
+    }
+
+    private void createEventListener(File configFile)
+    {
+        log.info("-- Loading event listener %s --", configFile);
+        if (configFile.exists()) {
+            configFile = configFile.getAbsoluteFile();
+            log.info("-- Loading event listener configFile.getAbsoluteFile() : %s --", configFile);
+            Map<String, String> properties;
+            try {
+                properties = new HashMap<>(loadProperties(configFile));
```

Comment:
This map doesn't seem to be used out of the try block,
```suggestion
            try {
                Map<String, String> properties = new HashMap<>(loadProperties(configFile));
```

Reply:
Done